test/openssl/test_ssl.rb

```diff
@@ -804,6 +804,71 @@ def socketpair
     end
   end
 
+  def test_keylog_cb
+    pend "Keylog callback is not supported" if !openssl?(1, 1, 1) || libressl?
+
+    possible_versions = [
+      OpenSSL::SSL::SSL3_VERSION,
+      OpenSSL::SSL::TLS1_VERSION,
+      OpenSSL::SSL::TLS1_1_VERSION,
+      OpenSSL::SSL::TLS1_2_VERSION
+    ]
```

Comment:
I think testing TLS 1.2 and TLS 1.3 is sufficient. It can make the test code cleaner (no need for `ignore_listener_error: true` or `rescue` clauses).

In ruby/openssl's tests, we make the implicit assumption that TLS 1.2 is always supported. You can use `tls13_supported?` helper method to see if TLS 1.3 is available or not.

Reply:
Thanks! I refactored the tests in [9f4fa70](https://github.com/ruby/openssl/pull/536/commits/9f4fa7081876ff3f6dad140da51e9f3b6240bc20)